pepper-apis/ddp-common/src/main/java/org/broadinstitute/ddp/service/FileDownloadService.java

```diff
@@ -0,0 +1,86 @@
+package org.broadinstitute.ddp.service;
+
+import com.google.auth.ServiceAccountSigner;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.cloud.storage.HttpMethod;
+import com.typesafe.config.Config;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.ddp.client.GoogleBucketClient;
+import org.broadinstitute.ddp.constants.ConfigFile;
+import org.broadinstitute.ddp.exception.DDPException;
+import org.broadinstitute.ddp.util.ConfigUtil;
+import org.broadinstitute.ddp.util.GoogleCredentialUtil;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+public class FileDownloadService {
+    private final ServiceAccountSigner signer;
+    private final GoogleBucketClient storageClient;
+    private final int maxSignedUrlMins;
+    private final String defaultDownloadBucket;
+
+    //GCP Service Account for file download
+    public static final String FILE_DOWNLOAD_CREDENTIALS = "fileDownload.signerServiceAccount";
```

Comment:
Nice! I'm planning on using the same configuration for the signerServiceAccount in DSM.